### ISSUE TITLE
Failed to compile lambda_http example code

### ISSUE THREAD

**Issue by Observer42** [awslabs/aws-lambda-rust-runtime#97]
I got below error when building the lambda_http example code

```rust
use lambda_http::{lambda, IntoResponse, Request, RequestExt};
use lambda_runtime::{Context, error::HandlerError};

fn main() {
    lambda!(hello)
}

fn hello(
    request: Request,
    _ctx: Context
) -> Result<impl IntoResponse, HandlerError> {
    Ok(format!(
        "hello {}",
        request
            .query_string_parameters()
            .get("name")
            .unwrap_or_else(|| "stranger")
    ))
}
```
according to issue #87 , I chose to fetch lambda_http crate from git address
the dependencies in Cargo.toml are:
```toml
lambda_runtime = "0.2"
lambda_http = {git = "https://github.com/awslabs/aws-lambda-rust-runtime"}
```

error message:
```
type mismatch in function arguments

expected signature of `fn(http::request::Request<lambda_http::body::Body>, lambda_runtime_core::context::Context) -> _`

note: required because of the requirements on the impl of `lambda_http::Handler<_>` for `fn(http::request::Request<lambda_http::body::Body>, lambda_runtime_core::context::Context) -> std::result::Result<impl lambda_http::response::IntoResponse, lambda_runtime_errors::HandlerError> {hello}`
note: required by `lambda_http::start`rustc(E0631)
<::lambda_http::lambda macros>(1, 27): expected signature of `fn(http::request::Request<lambda_http::body::Body>, lambda_runtime_core::context::Context) -> _`
main.rs(8, 1): found signature of `fn(http::request::Request<lambda_http::body::Body>, lambda_runtime_core::context::Context) -> _`
```

It will not receive such error If I change the lambda_http version to 0.1.0
But in a more practical project, serde::Deserialize (#87 ) will be introduced, so I cannot choose 0.1.0

Is there any workaround? or is there any example to return the http response via lambda_runtime crate only (I would prefer this way, because `lambda` macro in runtime crate allow custom Error type which implements `Fail + LambdaErrorExt + Display` (though still not convenient, i.e. #94 ), while `lambda` in http crate allow `HandlerError` only)? Thanks in advance

**Comment by softprops** [comment]
@davidbarsky would this issue be resolved if the http lambda crate were published today as is independently of the other crates?

**Comment by Observer42** [comment]
update: seems `lambda_http::start` is the wrapper of `lambda_runtime::start`, and `(http) Response / Request` can be transformed to internal struct `LambdaResponse / LambdaRequest`.

Currently my workaround is to create two structs similar to `LambdaRequest / LambdaResponse`, and put them in `Event` and `Output` in trait `lambda_runtime::Handler`. So I don't need to introduce lambda_http crate

**Comment by davidbarsky** [comment]
Sorry for the delay, I’m catching up now. @Softprops—maybe?

**Comment by softprops** [comment]
I can try some experiments and see

**Comment by bburnichon** [comment]
> @davidbarsky would this issue be resolved if the http lambda crate were published today as is independently of the other crates?

I cloned repository locally, changed Cargo.toml from lambda-http as follows:

`~/Github/awslabs/aws-lambda-rust-runtime/lambda-http (master *=)$ git diff`

```diff
diff --git a/lambda-http/Cargo.toml b/lambda-http/Cargo.toml
index e8e6d52..c8b147d 100644
--- a/lambda-http/Cargo.toml
+++ b/lambda-http/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "lambda_http"
-version = "0.1.0"
+version = "0.2.0"
 authors = ["Doug Tangren"]
 edition = "2018"
 description = "Rust API Gateway proxy event interfaces for AWS Lambda"
@@ -20,7 +20,7 @@ http = "0.1"
 serde = "^1"
 serde_json = "^1"
 serde_derive = "^1"
-lambda_runtime = { path = "../lambda-runtime", version = "^0.2" }
+lambda_runtime = "0.2"
 tokio = "^0.1"
 base64 = "0.10"
 failure = "0.1"
```

And then changed my Cargo.toml to reference this version instead of the one in crates.io:
```diff
diff --git a/Cargo.toml b/Cargo.toml
index 402fef2..e31644e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,7 +24,7 @@ actix-web = { optional = true, version = "0.7" }
 env_logger = { optional = true, version = "0.6" }
 listenfd = { optional =  true, version = "0.3"}
 lambda_runtime = { optional = true, version = "0.2" }
-lambda_http = { optional = true, version = "0.1" }
+lambda_http = { optional = true, version = "0.2", path = "/home/bburnichon/Github/awslabs/aws-lambda-rust-runtime/lambda-http" }
 #lambda_http = { optional = true, git = "https://github.com/awslabs/aws-lambda-rust-runtime" }
 simple-error = { optional = true, version = "0.1.13" }
```

I can now compile the example without the error message that @Observer42 obtained.

So I would say that publishing crates independently would solve the issue.

**Comment by softprops** [comment]
@bburnichon awesome work! 

I still had the issue mentioned above when depending on the latest git version. I think @bburnichon's trick looks promising: a temporary forced declaration of dependence on the published form of lambda_runtime instead of the latest source.

**Comment by softprops** [comment]
@davidbarsky thoughts on trying this to publishing lambda-http with a version dependency on `lambda_runtime = "0.2"`?

**Comment by softprops** [comment]
that would have a nice side effect of on the next update they all versions end up lining up to 0.3 or whatever they might be

**Comment by softprops** [comment]
@davidbarsky sending another ping :)

**Comment by davidbarsky** [comment]
@softprops Sorry about the delay—normal work has been a bit much recently. I think that's a good idea—I'll try to get a release out later today.

**Comment by jain98** [comment]
Issue seems to have been fixed in release [(#103)](https://github.com/awslabs/aws-lambda-rust-runtime/pull/103).
